src/gamecontroller/gamecontroller.h

```diff
@@ -81,6 +81,7 @@ class GameController : public InputDevice
     QHash<int, int> const &getDpadvalues();
 
     SDL_GameController *getController() const;
+    virtual int getControllerType() const override;
```

Comment:
Why you use type `int` instead of `SDL_GameControllerType`?

Reply:
SDL_GameControllerType is not available on SDL versions before 2.0.16.

Reply:
What do you think about this hack?
```cpp
#if SDL_VERSION_ATLEAST(2, 0, 16)
#else
typedef enum
{
    SDL_CONTROLLER_TYPE_UNKNOWN = 0,
} SDL_GameControllerType;
#endif
```

Reply:
Done.